include/envoy/buffer/buffer.h

```diff
@@ -287,6 +289,42 @@ class Instance {
 
     return fromEndianness<Endianness>(static_cast<T>(result)) | sign_extension_bits;
   }
+  /**
+   * A non-exception throwing replica of peekInt()
+   * This is created because the call site of peekInt() is large and the change requires
+   * more than a single PR. 
+   */ 
+  template <typename T, ByteOrder Endianness = ByteOrder::Host, size_t Size = sizeof(T)>
+  StatusOr<T> peekIntNoExcept(uint64_t start = 0) {
```

Comment:
drive by: instead of large changes like this can't you just throw the exception from the cc file? I don't see why these changes are necessary. Actually removing exceptions from control flow is orthogonal. Let's please keep the changes focused on just removing exceptions from header files which generally should be trivial.

Reply:
Working with template functions is a little different as we will need to do explicit template instantiations to move the template function implementations into .cc files. That also means that we need to instantiate all the currently used type in .cc files as such: 
```
template a_lot_of_types peekInt(uint64_t start);
```
But yeah, if you believe this way is better than what my current method is, I am certainly on board with it as it makes the change much narrower and cleaner (instead of going around changing all call sites).

Reply:
I don't even think that is necessary. Just define a non-templated helper which throws the exception and implement it in a cc file. This will fix the immediate problem if no throwing from header files very quickly and get the linter up. Now, whether this fixes the real problem in the sense that you can't through *through* header files is a different issue, but I would cross that bridge when you come to it.

Reply:
+1

Reply:
I think that is a great way to do this! Wonder why that didn't become obvious to me. But I have just heard that @asraa is trying to get rid of the exception throwing behavior altogether in the dataplane pipeline. So maybe she has some input on this topic.

But this is definitely a much better way of doing this.

Reply:
> But I have just heard that @asraa is trying to get rid of the exception throwing behavior altogether in the dataplane pipeline. So maybe she has some input on this topic.

As I said above, I think removing exceptions from certain code paths is orthogonal to this effort. My advice is to start by meeting your immediate goal in as simple a way as possible and then we can go from there?

Reply:
agreed.